File: ros/ws02/first_ws/src/use_custom_msg_pkg/scripts/publisher_cus_msg.py
#! /usr/bin/env python
# coding:utf8

import rospy
from custom_msgs.msg import Student

if __name__ == '__main__':

    nodeName = "publisher_node"
    rospy.init_node(nodeName)

    topicName = "/zcb01/topic"
    pub = rospy.Publisher(topicName, Student, queue_size=1000)

    rate = rospy.Rate(2)
    msg = Student()
    idx = 0
    while not rospy.is_shutdown():
        msg.name = "tom%d" % idx
        msg.age = idx
        pub.publish(msg)

        idx += 1
        rate.sleep()

File: ros/ws02/first_ws/src/use_custom_msg_pkg/src/publisher_cus_msg.cpp
//
// Created by zcb on 5/3/21.
//

#include <ros/ros.h>
#include <iostream>
#include <custom_msgs/Student.h>


int main(int argc, char **argv) {

    std::string nodeName = "publisher_node";
    ros::init(argc, argv, nodeName);
    ros::NodeHandle node;

    std::string topicName = "/zcb01/topic";
    ros::Publisher publisher = node.advertise<custom_msgs::Student>(topicName, 1000);

    custom_msgs::Student msg;

    ros::Rate rate(2);
    int idx = 0;
    while (ros::ok()) {
        msg.name = "tom" + std::to_string(idx);
        msg.age = idx;

        publisher.publish(msg);

        idx++;
        rate.sleep();
    }

    return 0;
}
